### PR TITLE
[Php80] Mirror additional docblock on importNames() on ClassPropertyAssignToConstructorPromotionRector

### LINKED ISSUE #7198: PHPDoc is sometimes lost when converting to constructor property promotion
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | 0.3.12           |

For context, I'm trying to convert the `doctrine/orm` codebase to PHP8 syntax. In [this PR](https://github.com/doctrine/orm/pull/9800), [the first commit](https://github.com/doctrine/orm/pull/9800/commits/4ebc979039af9ef7bcf8d6702a883b81effe1d85) was generated by Rector.

[The second commit](https://github.com/doctrine/orm/pull/9800/commits/6f3cbc3ae9f53d1a4f3d3ab443d3fcf5027566a2) shows many occurrences of this bug.

Here is an excerpt of the diff:

```diff
 class Action
 {
-    /**
-     * @var string
-     * @Id
-     * @Column(type="string", length=255)
-     * @GeneratedValue(strategy="NONE")
-     */
-    public $name;
-
     /**
      * @psalm-var Collection<int, Token>
      * @OneToMany(targetEntity="Token", cascade={"persist", "remove"}, mappedBy="action")
      */
     public $tokens;

-    public function __construct($name)
+    public function __construct(public $name)
     {
-        $this->name   = $name;
         $this->tokens = new ArrayCollection();
     }
```

## Minimal PHP Code Causing Issue

https://getrector.org/demo/64dd55dd-13ee-4d90-a852-ddaa8c37bc46

The call to `importNames()` seems to be important, and the fact that the phpdoc has several annotations too

## Expected Behaviour

The phpdoc should be preserved (for `@Id`), and converted to a native type (for `string`).

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/7198

### CODE REVIEW

**Comment by samsonasik** [comment]
The original non-autoimport needs to be mirror for `@var` to `@param` as well

**Comment by samsonasik** [comment]
Fixed 🎉 /cc @greg0ire

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules-tests/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector/Fixture/mirror_additional_docblock.php.inc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+use Doctrine\ORM\Mapping\Id;
+
+class MirrorAdditionalDocblock
+{
+    /**
+     * @param string $name
+     */
+    public function __construct(
+        /**
+         * @Id
```
Comment (greg0ire) [inline rules-tests/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector/Fixture/mirror_additional_docblock.php.inc:36]: I'd expect there to be either `@var string` here, or `public string $name` below, but I might be wrong :thinking:

Comment (samsonasik) [reply rules-tests/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector/Fixture/mirror_additional_docblock.php.inc:36]: No, the `@var` already moved to `@param` to fill the param doc type.

If the `property` is `non-typed`, and the `param` is `non-typed` as well, this rule is not changing to typed.

Comment (greg0ire) [reply rules-tests/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector/Fixture/mirror_additional_docblock.php.inc:36]: I see, that's the role of another rule I must be using 👍

Comment (samsonasik) [reply rules-tests/Php80/Rector/Class_/ClassPropertyAssignToConstructorPromotionRector/Fixture/mirror_additional_docblock.php.inc:36]: You can use `TypedPropertyRector` with set `INLINE_PUBLIC` to true to make property typed for public property:

```php
use Rector\Php74\Rector\Property\TypedPropertyRector;
// ...
    $rectorConfig
        ->ruleWithConfiguration(TypedPropertyRector::class, [
            // Set to false if you use in libraries, or it does create breaking changes.
            TypedPropertyRector::INLINE_PUBLIC => true,
        ]);
```

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Looks good 😊

**Pull request merged**